tasks/main.yml

```diff
@@ -1,34 +1,34 @@
 ---
 # tasks file for ansible-role-keepalived
-- name: "Include {{ ansible_distribution }} variables"
+- name: Load OS specific variables
   include_vars: "{{ item }}"
   with_first_found:
     - "{{ ansible_distribution }}.yml"
     - "{{ ansible_os_family }}.yml"
 
-- name: 'Install packages'
+- name: Install packages
   package:
-    name: "{{ item }}"
+    name: "{{ keepalived_packages }}"
     state: present
-  with_items: "{{ keepalived_packages }}"
-  register: install_packages
-  until: install_packages is success
-  retries: 5
-  delay: 2
 
-- name: 'Set configuration'
-  template:
-    src: "{{ keepalived_conf_template }}"
-    dest: "{{ keepalived_conf_path }}"
-    owner: root
-    group: root
-    mode: 0644
-  notify: reload keepalived
+- name: Allow binding non-local IPs to the hosts
+  sysctl:
+    name: net.ipv4.ip_nonlocal_bind
+    value: 1
+    ignoreerrors: yes
 
-- name: 'Start service'
+- name: Start and enable keepalived
   service:
     name: "{{ keepalived_service_name }}"
     state: started
     enabled: true
-  when: not keepalived_provision_only
+
+- name: Configure keepalived
```

Comment:
Starting a service that is not configured has a lot of caveats.
Would be better to check if the service is running, configure it then either start or reload depending on the return of the first check.

Reply:
So you mean I should revert the order of these two tasks, right? Anyway TIL I should configure a service before starting and enabling it. I've always done it the wrong way.

Reply:
Yes, that's right. That's the way we do it at least.